src/app/services/vm/vm.service.ts

```diff
@@ -50,18 +52,40 @@ export class VmService {
   });
 
   private ConsoleApiUrl;
+  private apiUrl;
 
   constructor(
     private http: HttpClient,
     private settings: ComnSettingsService,
     private titleService: Title
   ) {
     this.ConsoleApiUrl = settings.settings.ConsoleApiUrl + 'vms/vsphere/';
+    this.apiUrl = settings.settings.ConsoleApiUrl + 'vms/';
     this.model = new VmModel();
   }
 
-  public getVm(id: string) {
-    return this.http.get(this.ConsoleApiUrl + id).pipe(
+  public getVm(id: string): Observable<VmModel> {
+    return this.http.get<VmModel>(this.ConsoleApiUrl + id).pipe(
+      catchError((error: any) => {
```

Comment:
What's the use case for the various if else statements. seems to me that it could be simplified to
`catchError((error) => observableThrowError(new Error(error.status)))`

Reply:
No reason, just was copying what was already there, but you're right it's a little weird. I'll probably refactor all of this to use open-api-generator in the next PR so this shouldn't be an issue.